src/lib/forms/core/PriceField.js

```diff
@@ -53,20 +53,20 @@ class PriceDropdown extends Component {
   };
 
   getAllOptions = (options, value) => {
-    const {
-      field: { required },
-    } = this.props;
+    const { defaultCurrency } = this.props;
     const { isLoading } = this.state;
-    if (!required) {
-      options = [
-        {
-          key: '',
-          value: undefined,
-          text: '-',
-        },
-        ...options,
-      ];
-    }
+    const otherOptions =
+      defaultCurrency !== null
+        ? options.filter(o => o.value === defaultCurrency)
+        : options;
+    options = [
+      {
+        key: '',
+        value: '',
+        text: '-',
+      },
+      ...otherOptions,
+    ];
```

Comment:
Instead of redefining `options` which is our argument we can do something like
```suggestion
    otherOptions.unshift({key: '', value: '', text: '-' })
```

Reply:
I'm also not a fan of reassigning a value to the argument.